src/js/button.js

```diff
@@ -7,13 +12,13 @@
  * @class
  * @constructor
  */
-vjs.Button = vjs.Component.extend({
+var Button = Component.extend({
   /**
    * @constructor
    * @inheritDoc
    */
   init: function(player, options){
```

Comment:
I maybe wrong but, since you proposed to use es6 classes that means that you will use constructor, extend and super to build your components. right? 
Wouldn't that change the way we create the components?? Even if we don't migrate to es6 the component?
For example this component below:
```
vjs.FullWidthPopup = vjs.Component.extend({
  /** @constructor */
  init: function (player, options) {
    vjs.Component.call(this, player, options);
    var closeButton = new vjs.CloseButton(player, options);
    this.addChild(closeButton);
    this.player_.on('CloseButton.pressed', vjs.bind(this, function (){
      this.trigger('FullWidthPopup.close');
    }));
  }
});

vjs.FullWidthPopup.prototype.createEl = function(type, props) {
  props = props || {};
  props.className = (props.className || '') + ' vjs-full-width-popup vjs-control';
  return vjs.Component.prototype.createEl.call(this, type || 'div', props);
};
```
How would you migrate it??